rails6/fr/chapter09-optimization.adoc

```diff
@@ -203,10 +203,10 @@ class Api::V1::OrdersControllerTest < ActionDispatch::IntegrationTest
 
     json_response = JSON.parse(response.body)
     assert_equal @order.user.orders.count, json_response['data'].count
-    assert_not_nil json_response.dig(:links, :first)
-    assert_not_nil json_response.dig(:links, :last)
-    assert_not_nil json_response.dig(:links, :prev)
-    assert_not_nil json_response.dig(:links, :next)
+    assert_not_nil json_response['links']['first']
+    assert_not_nil json_response['links']['last']
+    assert_not_nil json_response['links']['prev']
+    assert_not_nil json_response['links']['next']
```

Comment:
This may produce `NoMethodError`. Here an example:

~~~rb
2.6.5 :001 > require 'json'
2.6.5 :002 > data = JSON.parse '{ "a": { "b": { "c" : 1 } } }'
2.6.5 :006 > data[:x][:y][:z]
Traceback (most recent call last):
        7: from /home/alexandre/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/alexandre/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/alexandre/.rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/alexandre/.rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/alexandre/.rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):06
        1: from (irb):06:in `rescue in irb_binding'
NoMethodError (undefined method `[]' for nil:NilClass)
~~~